apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
-        },
-      );
-    }
-  } else if (req.body.__action === 'delete_question') {
-    debug('Delete question');
-    const editor = new QuestionDeleteEditor({
-      locals: res.locals,
-    });
-    editor.canEdit((err) => {
-      if (ERR(err, next)) return;
-      editor.doEdit((err, job_sequence_id) => {
-        if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-          res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-        } else {
-          res.redirect(res.locals.urlPrefix + '/course_admin/questions');
-        }
+        });
+      }
+    } else if (req.body.__action === 'delete_question') {
+      const editor = new QuestionDeleteEditor({
+        locals: res.locals,
       });
-    });
-  } else if (req.body.__action === 'sharing_set_add') {
-    debug('Add question to sharing set');
-    features
-      .enabledFromLocals('question-sharing', res.locals)
-      .then((questionSharingEnabled) => {
-        if (!questionSharingEnabled) {
-          return next(error.make(403, 'Access denied (feature not available)'));
-        }
-        if (!res.locals.authz_data.has_course_permission_own) {
-          return next(error.make(403, 'Access denied (must be a course Owner)'));
-        }
-        sqldb.queryZeroOrOneRow(
-          sql.sharing_set_add,
-          {
+      editor.assertCanEdit();
+      await editor.doEditAsync();
+      res.redirect(res.locals.urlPrefix + '/course_admin/questions');
+    } else if (req.body.__action === 'sharing_set_add') {
+      features
+        .enabledFromLocals('question-sharing', res.locals)
+        .then((questionSharingEnabled) => {
+          if (!questionSharingEnabled) {
+            throw error.make(403, 'Access denied (feature not available)');
+          }
+          if (!res.locals.authz_data.has_course_permission_own) {
+            throw error.make(403, 'Access denied (must be a course Owner)');
+          }
+          sqldb.queryAsync(sql.sharing_set_add, {
             course_id: res.locals.course.id,
             question_id: res.locals.question.id,
             unsafe_sharing_set_id: req.body.unsafe_sharing_set_id,
-          },
-          (err) => {
-            if (ERR(err, next)) return;
-            res.redirect(req.originalUrl);
-          },
-        );
-      })
-      .catch((err) => next(err));
-  } else if (req.body.__action === 'share_publicly') {
-    debug('Add question to sharing set');
-    features
-      .enabledFromLocals('question-sharing', res.locals)
-      .then((questionSharingEnabled) => {
-        if (!questionSharingEnabled) {
-          return next(error.make(403, 'Access denied (feature not available)'));
-        }
-        if (!res.locals.authz_data.has_course_permission_own) {
-          return next(error.make(403, 'Access denied (must be a course Owner)'));
-        }
-        sqldb.queryZeroOrOneRow(
-          sql.update_question_shared_publicly,
-          {
+          });
+        })
+        .then(() => {
```

Comment:
Now that the enclosing function is async, there's not need to use `.then()` and `.catch()`; just use normal control flow with `await`. Ditto below for `share_publicly`.

Reply:
Updated here and `share_publicly` to use `await` instead of `.then()/.catch()` in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).